### PR TITLE
Add getsentry setup and storybook content

### PR DESCRIPTION
**Pull request by markstory**
Migrate this content from notion to here. Once my CI changes are merged
I'll migrate the staging environment docs.

### CODE REVIEW

File: src/docs/environment.mdx [hunk 1]
```diff
@@ -170,6 +172,106 @@ If you would like to be able to run `devserver` outside of your root checkout, y
...
+
+```bash
+# Go to where you have sentry and clone getsentry.
+cd /path/to/sentry
+cd ..
+git clone git@github.com:getsentry/getsentry.git
+```
+
+It's necessary to keep getsentry in an adjacent directory (it's expected by
+`install-develop-requirements` in the getsentry Makefile). For example, if you
```
Comment (joshuarli) [inline src/docs/environment.mdx:196]: s/install-develop-requirements/install-py-dev

File: src/docs/environment.mdx [hunk 2]
```diff
@@ -170,6 +172,106 @@ If you would like to be able to run `devserver` outside of your root checkout, y
...
+getsentry, the settings are configured in `getsentry/conf/settings/defaults.py`.
+Additional environment specific overrides are found in files matching the
+environment name. For example, local development is found in
+`getsentry/conf/settings/dev.py`.
+
+For sentry, the settings file is installed at `~/.sentry/sentry.conf.py`
+(outside both repos). If you're investigating, or want to explore, discrepancies
+between the two files, those two files are a good place to start looking.
+
+If you're in your getsentry virtualenv, do a `pip freeze | grep sentry\@` and
```
Comment (joshuarli) [inline src/docs/environment.mdx:248]: Everything from this line down can be removed. I wrote it all, but it's just too verbose of an explanation of what an editable install is, and it's basic knowledge anyways.

File: src/docs/environment.mdx [hunk 3]
```diff
@@ -170,6 +172,106 @@ If you would like to be able to run `devserver` outside of your root checkout, y
...
+
+# Initialize and bootstrap the database
+make reset-db
+
+# Run the development server
+getsentry devserver --workers
+```
+
+Once you've booted the server you'll access the development environment
+via [http://dev.getsentry.net:8000](http://dev.getsentry.net:8000/), which will
```
Comment (joshuarli) [inline src/docs/environment.mdx:226]: I don't have enough context on this, but... I feel like this introduces unnecessary friction? I'd expect getsentry vs sentry dev to be pretty much the same, so getsentry devserver should be accessible via `localhost:8000`. cc @dcramer

File: src/docs/environment.mdx [hunk 4]
```diff
@@ -170,6 +172,106 @@ If you would like to be able to run `devserver` outside of your root checkout, y
...
+
+It's necessary to keep getsentry in an adjacent directory (it's expected by
+`install-develop-requirements` in the getsentry Makefile). For example, if you
+did a `ls ~/code` you'd see something like:
+
+```bash
+sentry/   getsentry/
+```
+
+Next, create a virtualenv for `getsentry`, and install its dependencies:
```
Comment (joshuarli) [inline src/docs/environment.mdx:203]: I would replace all this with simply: Run `make develop` and follow the instructions from there.

Reason being it's redundant now that `make develop` will hold your hand:

```
You don't seem to have a virtualenv. Please create one by running: python -m virtualenv .venv
make: *** [ensure-venv] Error 1
```

File: src/docs/environment.mdx (hunk 3 above)
Comment (markstory) [reply src/docs/environment.mdx:226]: The current getsentry configuration has `PARENT_HOST` set to dev.getsentry.net, and `options[system.url-prefix] =  "dev.getsentry.net:8000"` so I think this is required right now. We could streamline that process in the future though.

File: src/docs/frontend/storybook.mdx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+---
+title: Storybook Styleguide
```
Comment (joshuarli) [inline src/docs/frontend/storybook.mdx:2]: What's leftover in the Notion doc:

```
## **Relevant Resources**

- Feature request: [https://github.com/getsentry/ops-req/issues/35](https://github.com/getsentry/ops-req/issues/35)
- PR with implementation: [https://github.com/getsentry/sentry/pull/8412](https://github.com/getsentry/sentry/pull/8412)
```

@billyvg is this important information?

Comment (billyvg) [reply src/docs/frontend/storybook.mdx:2]: Hmm not really, I mean there's some good information in the implementation PR

**Pull request merged**